skillmap/src/lib/workspaceProvider.ts

```diff
@@ -20,10 +23,26 @@ export async function saveProjectAsync(project: pxt.workspace.Project): Promise<
     await ws.saveProjectAsync(project);
 }
 
-export async function getUserStateAsync(): Promise<UserState | undefined> {
+export async function getUserStateAsync(): Promise<UserState> {
     const ws = await getWorkspaceAsync();
 
-    return ws.getUserStateAsync();
+    let userState = await ws.getUserStateAsync();
+    if (!userState) {
+        userState = {
+            id: guidGen(),
+            completedTags: {},
+            mapProgress: {},
+            version: pxt.skillmap.USER_VERSION
+        };
+    }
+    // Read synchronized skillmap state from cloud profile.
+    const skillmapState = await authClient.getSkillmapStateAsync();
+    userState = {
```

Comment:
this should use the "version" from the authClient

Reply:
Can you expand? The meaning of the field hasn't changed, just moved where it is initialized.

Reply:
@riknoll My sense is that skillmap should own this version number. Does that square with your thinking?